yew/src/lib.rs

```diff
@@ -98,6 +98,70 @@
 #![recursion_limit = "512"]
 extern crate self as yew;
 
+/// This macro helps you manage a list of HTML classes.
+///
+/// This macro is similar to [`classNames`](https://www.npmjs.com/package/classnames) from
+/// JavaScript. It helps you manage a set of classes more easily.
+///
+/// # Examples
```

Comment:
```suggestion
/// This macro provides a convenient way to create [`Classes`].
///
/// The macro takes a list of items similar to the [`vec!`] macro and returns a [`Classes`] instance.
/// Each item can be of any type that implements `Into<Classes>` (See the implementations on [`Classes`] to learn what types can be used).
///
/// # Example
```

Reply:
:thinking: 

> /// This macro provides a convenient way to create [`Classes`].

I prefer my sentence. My reasoning is that `Classes` is kinda of an implementation detail. It's not like you are going to create a `Classes` object and manipulate it a lot: most of the time you just give all your classes to the macro, then pass the result to `class=` on some node, and that's it.

> /// The macro takes a list of items similar to the [`vec!`] macro and returns a [`Classes`] instance.
> /// Each item can be of any type that implements `Into<Classes>` (See the implementations on [`Classes`] to learn what types can be used).

That is more detail indeed but I like the comparison I did with classNames.

classNames is very well known among JS developers (which is probably a good part of yew's community). Giving a good example is sometimes better than a long description.

> Not perfectly happy with my suggestion but at least it goes into more detail.

I think having more details is good. But if you don't mind I think it's important to keep the comparison with classNames and the more pragmatic description "manage HTML classes".

Reply:
> My reasoning is that Classes is kind of an implementation detail

And that's where we disagree. `Classes` is (and should be) part of the public API, if you disagree with that then we need to deal with that separately.
I do agree with "It's not like you are going to create a Classes object and manipulate it a lot"! I really don't like that the current implementation of `Classes` is optimized for fast manipulation rather than creation and turning into an attribute. If you want to change that, I'm all for it, but as long as the former remains true our API reference should accurately reflect that. Also, if we don't tie this to `Classes` at all we don't have a way to easily explain the return type of the macro (because users are allowed to use it outside of the `html!` macro) and what types of arguments it accepts.

> classNames is very well known among JS developers (which is probably a good part of yew's community).

I totally get it but please imagine the situation if you don't know about the library. It really comes across like "if you don't know what this is, look at the documentation of this other library written for another language". There are (surprisingly) many developers using Yew that aren't familiar with the Javascript ecosystem and as I said previously, those who are will probably recognize the similarities by themselves.

The comparison isn't even a particularly good one because they aren't 100% equal. Sure `classes!` vaguely shares the same idea of combining multiple strings into a single construct but that's about as far as it goes currently.
If you really want to, for some reason, I can just barely live with putting it on the website, but please don't put it in the docstring.

> I think having more details is good. But if you don't mind I think it's important to keep the comparison with classNames and the more pragmatic description "manage HTML classes".

The reason I replaced "manage HTML classes" is because the macro doesn't do any "managing" to speak of. It always just creates a new `Classes` instance. It's totally fine to use something like this on the website because that's supposed to serve as an introduction but the docstrings are an API reference and there is an expectation for them to be accurate and complete.

Reply:
:thinking: all fair points

Reply:
I was so worried I was coming on too strong. Thanks for being understanding!